Add <IfLocale> component

Ken-Lo opened the pull request:
RE: https://github.com/gocardless/splash-pages/issues/168

This PR adds a `<IfLocale>` component. You can conditionally show information based on the local properties in each country (e.g. based on the domestic DD scheme or whether instant sign-up is available in the country).

```jsx
<IfLocale domesticScheme='bacs'>
  Stuff about Bacs Direct Debit
</IfLocale>
<IfLocale domesticScheme='sepa'>
  Stuff about SEPA Direct Debit
</IfLocale>
<IfLocale domesticScheme='bankgirot'>
  Stuff about Bg Autogiro
</IfLocale>

<IfLocale hasInstantSignup>
  Signup now in 5 minutes
</IfLocale>
<IfLocale hasInstantSignup={false}>
  Contact Sales
</IfLocale>

<IfLocale hasPercentagePricing>
  % pricing is available
</IfLocale>
<IfLocale hasPercentagePricing={false}>
  % pricing is not available
</IfLocale>
```

In order to make it work, the `country_properties` message needs to be set for each locale in the app/messages/ folder. e.g.

```js
  country: 'United Kingdom',
  country_properties: {
    local_scheme: 'bacs',
    has_instant_signup: true,
    has_percentage_pricing: true,
  },
```

```js
  country: 'Belgique',
  country_properties: {
    local_scheme: 'sepa',
    has_instant_signup: false,
    has_percentage_pricing: true,
  },
```

Ken-Lo commented:
@harrison Would be good to get your thoughts on this approach.

Also thinking about possibly generalising this component to `<IfLocal>` and allowing it to work on other country-based properties (e.g. whether immediate sign-up is available, whether Basic pricing is available)

jackfranklin commented on app/components/if-local-scheme/if-local-scheme.js:
```diff
@@ -0,0 +1,39 @@
+    }
+  }
+
+  return localesWithScheme;
```
you could use `filter` from LoDash here:

```js
return _.filter(countryProperties, function(locale) {
  return locale.local_scheme === selectedScheme;
});
```

harrison replied:
:+1: 
but think you will need to return the object key:
```
return Object.keys(countryProperties).filter(function(locale) {
  return countryProperties[locale].local_scheme === selectedScheme;
});
```

harrison replied:
Wait do you need this?

Could you do this in render?

```js
const schemes = getMessage(messages, 'country_properties');

 if (schemes. local_scheme !== this.props.name) { return null; }
return React.createElement('span', this.props, this.props.children);
```

harrison replied:
think you could then remove `countryProperties`

Ken-Lo replied:
Nice one @jackfranklin, I knew there must have been a better way of selecting items from an array (didn't find anything in the MDN docs). Thanks for the info about lodash!

@harrison Makes sense, if we aren't wrapping `<Translation>`, then we wouldn't need to know the properties of other countries. Will move to render and send you a v2 :)

jackfranklin commented:
@Ken-Lo welcome to the frontend team! ;) This looks really awesome. I'll let @harrison review other than the one thing I spotted (happy to chat through further) but just wanted to say that this seems like a really nice, intuitive improvement that will help a lot :+1:

harrison commented on app/components/if-local-scheme/if-local-scheme.js:
```diff
@@ -0,0 +1,39 @@
+  }
+
+  render() {
+    const { availableLocales, countryProperties } = this.context;
```
`availableLocales` is unused

Ken-Lo commented:
@harrison Just updated this:

* Moved everything to render() so no need to pass countryProperties in the context
* Generalised to `<IfLocale>` so we can also do this on other country properties
* Added some tests
* Updated a couple of new pages to use the `<IfLocale>` tag

Would be good to get your thoughts!

harrison commented on app/components/if-locale/if-locale.spec.js:
```diff
@@ -0,0 +1,163 @@
+      });
+    });
+  });
+
```
:scissors: empty lines

harrison commented on app/components/if-locale/if-locale.spec.js:
```diff
@@ -0,0 +1,163 @@
+}
+
+describe('IfLocale Component', () => {
+
```
:scissors: empty lines

harrison commented:
Nice work! :+1: minor comments

harrison commented:
Closes #168

Ken-Lo commented:
@harrison Thanks! Merging this now, will tidy up the FAQ & other pages to use `<IfLocale>` in a separate pull request.
